src/main/scala/com/databricks/spark/csv/readers/readers.scala

```diff
@@ -23,6 +23,21 @@ import java.io.StringReader
 import com.univocity.parsers.csv._
 
 /**
+  * Allows for greater extensibility
+  */
+trait BulkReader extends Iterator[Array[String]] {
+  protected def getReader(iter: Iterator[String]) = new StringIteratorReader(iter)
```

Comment:
getters/setters are not really idiomatic in scala code - you can set this as a `val` on the trait